_posts/2019-04-16-accessing-kafka-part-1.md

```diff
@@ -0,0 +1,107 @@
+This is especially visible when running Kafka inside a platform like Kubernetes but is accessed from outside of the platform.
+This blog-series will explain how Kafka and its clients work and how Strimzi deals with it to make it accessible for clients running outside of Kubernetes.
+
+<!--more-->
+
+It would be of course not sufficient to just shard the data into partitions.
+Also the ingress and egress data traffic needs to be properly sharded.
+The clients writing or reading from given partition have to connect directly to the broker which is hosting it.
+Thanks to the clients connecting directly to the individual brokers, the brokers don't need to do any forwarding of data between the clients and the partition it self.
+That helps to significantly reduce the amount of work the brokers have to do and the amount of traffic flowing around within the cluster.
+The only data traffic between the different brokers is the replication, when the follower replicas are receiving their data from the leader replicas.
+That makes the data shards independent on each other and that makes Kafka scale so well.
+
+![Clients connecting to partitions]({{ "/assets/2019-04-16-connecting-to-leader.png" }})
+
+But how do the clients know where to connect?
+
+# Kafka's discovery protocol
+
+Kafka has its own discovery protocol.
+When a Kafka client is connecting to the Kafka cluster, it first connects to any broker which is member of the cluster and asks it for _metadata_ for one or more topics.
+These _metadata_ contain the information about the topics, its partitions and brokers which are responsible for them.
+All brokers should have these data for the whole cluster because they are all synced through Zookeeper.
+Therefore it doesn't matter to which broker the client connected as first - all of them will give it the same response.
+
+![Connection flow between Kafka client and Kafka cluster]({{ "/assets/2019-04-16-connection-flow.png" }})
+
+Once the client gets the _metadata_, it will use them to figure out where to connect when it wants to write to or read from given partition.
+The broker addresses used in the _metadata_ will be either created by the broker it self based on the hostname of the machine where the broker runs.
+Or it can be configured by the used using the `advertised.listeners` option.
+The client will use the address from the _metadata_ to open one or more new connections to the addresses of the brokers which hosts the particular partitions it is interested in.
+And these connection will be used to produce or consume data.
+
+_Note: The description of the Kafka protocol is intentionally simplified for the purpose of this blog post._
+
+# What does it mean for Kafka on Kubernetes
+
+So, what does this mean for running Kafka on Kubernetes?
+If you are familiar with Kubernetes, you probably know that the most common way how to expose some application is using a Kubernetes service.
```

Comment:
```suggestion
If you are familiar with Kubernetes, you probably know that the most common way expose some application is using a Kubernetes `Service`.
```

Reply:
Should we keep the `to` here? E.g. `common way to expose`?